chain-impl-mockchain/src/ledger/evm.rs

```diff
@@ -21,21 +24,43 @@ use thiserror::Error;
 #[derive(Debug, Error, Clone, PartialEq, Eq)]
 pub enum Error {
     #[error(
-        "for the provided jormungandr account: {} or evm account: {} mapping is already exist", .0.to_string(), .1.to_string()
+        "For the provided jormungandr account: {} or evm account: {} mapping is already exist", .0.to_string(), .1.to_string()
     )]
     ExistingMapping(JorAddress, EvmAddress),
-    #[error("evm transaction error: {0}")]
+    #[error("Canot map address: {0}")]
+    CannotMap(#[from] LedgerError),
+    #[error("EVM transaction error: {0}")]
     EvmTransaction(#[from] chain_evm::machine::Error),
-    #[error("Protocol evm mapping payload signature failed")]
-    EvmMappingSignatureFailed,
+    #[error("It is not a contranct generation transaction type")]
+    NotAContractType,
 }
 
-#[derive(Clone, PartialEq, Eq)]
+#[derive(Debug, Clone, PartialEq, Eq)]
 pub struct AddressMapping {
     evm_to_jor: Hamt<DefaultHasher, EvmAddress, JorAddress>,
     jor_to_evm: Hamt<DefaultHasher, JorAddress, EvmAddress>,
 }
 
+/// One way transforming procedure from `EvmAddress` to `JorAddress`.
+/// This allows to map any `EvmAddress` to the `JorAddress` before explicit execution of the `EvmMapping` transaction.
+/// Intention - is to have possibility to link an EVM Contarct account with a Jormungandr account.
```

Comment:
those are two slightly different use cases.
1) standard account not yet mapped, but that _could_ be mapped in the future
2) contract account that can't possibly be mapped